File: c++/5.cpp
#include <iostream>
#include <algorithm>
#include <string>

bool isPalindrome(int x) {
  std::string numberString = std::to_string(x);
  std::string reversedString = numberString;
  std::reverse(reversedString.begin(), reversedString.end());
  return numberString == reversedString;
}

int main() {
  std::cout << "\n\n\n" << isPalindrome(121);
  std::cout << isPalindrome(212);
  std::cout << isPalindrome(98) << "\n\n\n";
  return 0;
}

File: java/5.java
public class Solution {
  public static boolean isPalindrome(int x) {
    String numberString = Integer.toString(x);
    String reversedString = new StringBuilder(numberString).reverse().toString();
    return numberString.equals(reversedString);
  }

  public static void main(String[] args) {
    System.out.println("\n\n\n" + isPalindrome(121));
    System.out.println(isPalindrome(212));
    System.out.println(isPalindrome(98) + "\n\n\n");
  }
}
